### ISSUE TITLE
Bucket-type in MapReduce [JIRA: RIAK-1622]

### ISSUE THREAD

**Issue by AndreasMadsen** [basho/riak#615]
The MapReduce documentation indicated that Bucket Types are supported by MapReduce:  

> When you know the set of objects over which you want to MapReduce (i.e. the locations of the objects, as specified by bucket type, bucket, and key)

However there is no documentation for how to specific a bucket-type in a MapReduce query. If one create a malformed `inputs` parameter the following message is obtained.

```
Valid formats are:
Key List:
  Map/Reduce over the provided set of keys.
  {\"inputs\":[[\"mybucket\",\"mykey1\"], \"mybucket\",\"mykey2\"], ...]}
Key List with Key Data:
  Map/Reduce over the provided set of keys.
  {\"inputs\":[[\"mybucket\",\"mykey1\", \"mykeydata1\"], [\"mybucket\",\"mykey2\", \"mykeydata1\"], ...]}
Index Query - Exact Match:
  Indexing backend must be enabled.
  {\"inputs\":{\"bucket\":\"mybucket\", \"index\":\"myindex_bin\", \"key\":\"mykey\"}}
Index Query - Range Search:
  Indexing backend must be enabled.
  Use secondary indexes to generate a starting set of keys.
  {\"inputs\":{\"bucket\":\"mybucket\", \"index\":\"myindex_bin\", \"start\":\"key1\", \"end\":\"key2\"}}
Search Query:
  Riak Search must be enabled and bucket hook installed.
  {\"inputs\":{\"bucket\":\"mybucket\", \"query\":\"foo OR bar\"}}
Search Query With Filter:
  Riak Search must be enabled and bucket hook installed.
  {\"inputs\":{\"bucket\":\"mybucket\", \"query\":\"field1:(foo OR bar)\", \"filter\":\"field2:baz\"}}
Bucket:
  Map/Reduce over all keys in the provided bucket.
  WARNING: THIS CAN BE A SLOW OPERATION!
  {\"inputs\":\"mybucket\"}\n\n","Bucket With Key Filter:
  Filter keys in a bucket, then Map/Reduce.
  WARNING: THIS CAN BE A SLOW OPERATION!
  {\"inputs\":{\"bucket\":\"mybucket\", \"key_filters\":[[\"matches\", \"foo.*\"]]}}
```

Again, none of these contain any information about how to specify the bucket type. In an attempt to guess the format I try this:

``` javascript
{
    inputs: {
      type: 'mytype',
      bucket: 'mybucket',
      index: 'myindex_bin',
      key: '0'
    },
    query: [{
      map: {
        language: 'javascript',
        source: (function (value, keyData, arg) {
          return [value];
        }).toString()
      }
    }]
  }
```

Where I have a `mybucket` in the default bucket-type with 3 items, and a `mybucket` in the `mytype` bucket-type with 1 items. I get 3 items, thus I must conclude that using the `type` property is not correct.

tl:dr;
Do MapReduce support bucket-type if not, that should be more clear from the online documentation. If it is supported how do I specify the bucket-type in the MapReduce query.

I'm using Riak 2.0.1 on Mac OS X 10.10

**Comment by jpmitche11** [comment]
I agree this feature is not well documented.

I was able to get a mapReduce query to work by setting input as follows:
{
    inputs: [bucketType, bucketName],
    query: ...
}

**Comment by cmeiklejohn** [comment]
@lucperkins Can we get this added to the MapReduce documentation?  It just came up on the mailing list again, and appears to be a straightforward fix.

**Comment by joaonrb** [comment]
@jpmitche11 are you sure that you define bucket type like that? My query return 200 like that but do not return me the expected values. And according to documentation Erlang should verify a list with 2 strings to be "bucketName" and "key".

**Comment by mitchellwrosen** [comment]
@joaonrb I can confirm that the two-element list is interpreted as `[bucket_type, bucket_name]` rather than `[bucket_name, key]` (even though the documentation says otherwise).

**Comment by Basho-JIRA** [comment]
https://github.com/basho/basho_docs/pull/2430 

***[posted via JIRA by Lauren Rother]***

**Comment by Basho-JIRA** [comment]
KV 2.2.0 - https://github.com/basho/basho_docs/pull/2430

We're waiting on a PR to 2.0.8: https://github.com/basho/basho_docs/tree/riak_2.0.8/content/riak/kv/2.0.8 

***[posted via JIRA by Lauren Rother]***

**Comment by Basho-JIRA** [comment]
https://github.com/basho/basho_docs/pull/2440 PR into 2.0.8, Travis failed though 

***[posted via JIRA by Eric Johnson]***
